executor/instruction/mod.rs

```diff
@@ -358,29 +365,96 @@ impl<T: Hkt> Checker<T> {
+                            range = intersect(range, comp_range);
+                        }
+                    } else {
+                        debug_assert!(
+                            rhs.as_variable().expect("RHS of comparison must be a variable") == target_variable
+                        );
+                        let lhs_variable_value = Self::get_vertex_value(lhs, row.as_ref(), &context.parameters);
+                        let lhs_value =
+                            Self::read_value(context.snapshot.as_ref(), &context.thing_manager, &lhs_variable_value)?;
+                        if let Some(lhs_value) = lhs_value {
+                            let comp_range = match comparator {
+                                Comparator::Equal => (Bound::Included(lhs_value.clone()), Bound::Included(lhs_value)),
+                                Comparator::Less => (Bound::Excluded(lhs_value), Bound::Unbounded),
+                                Comparator::LessOrEqual => (Bound::Included(lhs_value), Bound::Unbounded),
+                                Comparator::Greater => (Bound::Unbounded, Bound::Excluded(lhs_value)),
+                                Comparator::GreaterOrEqual => (Bound::Unbounded, Bound::Included(lhs_value)),
+                                Comparator::Like => continue,
+                                Comparator::Contains => continue,
+                                Comparator::NotEqual => continue,
+                            };
+                            range = intersect(range, comp_range);
+                        }
+                    }
                 }
                 _ => (),
             }
         }
-        range
-        */
-        todo!() as (Bound<VariableValue<'_>>, Bound<VariableValue<'_>>)
+        let range = (range.0.map(|value| value.into_owned()), range.1.map(|value| value.into_owned()));
+        Ok(range)
+    }
+
+    fn get_vertex_value<'b>(
+        vertex: &'b CheckVertex<ExecutorVariable>,
+        row: Option<&'b MaybeOwnedRow<'b>>,
+        parameters: &'b ParameterRegistry,
+    ) -> VariableValue<'b> {
+        match vertex {
+            CheckVertex::Variable(var) => match var {
+                ExecutorVariable::RowPosition(position) => {
+                    row.expect("CheckVertex::Variable requires a row to take from").get(*position).as_reference()
+                }
+                ExecutorVariable::Internal(_) => {
+                    unreachable!("Comparator check variables must have been recorded in the row.")
+                }
+            },
+            CheckVertex::Type(type_) => VariableValue::Type(type_.clone()),
+            CheckVertex::Parameter(parameter_id) => {
+                VariableValue::Value(parameters.value_unchecked(*parameter_id).as_reference())
+            }
+        }
+    }
+
+    fn read_value<'b, 'a>(
+        snapshot: &'a impl ReadableSnapshot,
+        thing_manager: &'a ThingManager,
+        variable_value: &'a VariableValue<'a>,
```

Comment:
`'b` is unused, `VariableValue<'a>` is redundant and potentially overly restrictive. `'a` on snapshot may be redundant as well, but I can't tell without static analysis.
```suggestion
    fn read_value<'a>(
        snapshot: &'a impl ReadableSnapshot,
        thing_manager: &'a ThingManager,
        variable_value: &'a VariableValue<'_>,
```